### PR TITLE
Add a progress bar to show dataset progress

### PR DESCRIPTION
**Pull request by hudson-newey**
# Add a progress bar to show dataset progress

A short description of the pull request

### Features

- Add progress bar to verification grid to show how far through the dataset the user is
- Adds `aria-label` to classification component buttons (tested with screenreaders)

### Bug Fixes

- Moves the `1rem` `kbd` element height to the abstract decision component. This was done so that the fixed `kbd` element height is does not apply to the `kbd` element helpers shown during `Alt` + `#` selection or in the help dialog
- During Playwright tests, the spectrogram container used to grow indefinitely. This has now been correctly fixed meaning that tests are **much** less flakey, and easier to develop. It also fixes tests failing on the main branch

### Code Quality

None

## Visual Changes

image

_A progress bar when **not** viewing history_

---

image

_A progress bar **when** viewing history_

In this image, I am viewing history, the darker region of the progress bar shows the current position of my view head (what is being shown on the screen). While the less shaded region shows where my verification head is up to.

Note: Tooltips always show how far through your verification head is

## Related Issues

Fixes: #163

## Final Checklist

- [x] All commits messages are semver compliant
- [x] Added relevant unit tests for new functionality
- [x] Updated existing unit tests to reflect changes
- [x] Code style is consistent with the project guidelines
- [ ] Documentation is updated to reflect the changes (if applicable)
- [x] Link issues related to the PR
- [x] Assign labels if you have permission
- [x] Assign reviewers if you have permission
- [ ] Ensure that CI is passing
- [x] Ensure that `pnpm lint` runs without any errors
- [x] Ensure that `pnpm test` runs without any errors

### CODE REVIEW

**Review by atruskie — APPROVED** [review]
Looks great.  However we should consider what happens if the paging callback implemented does not provide a valid value to `totalItems`. What if it is 0 or negative? Or falsey? One way this (unbounded or incalculable totals) has been historically dealt with is a logarithmic function that never reaches 1, e.g.: https://www.wolframalpha.com/input?i=plot+0.93*log_10%281%2Bx*10%29+from+0+to+1

File: src/components/progress-bar/progress-bar.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ * @description
+ * A progress bar that indicates how far through a task you are.
+ */
+@customElement("oe-progress-bar")
+export class ProgressBar extends AbstractComponent(LitElement) {
+  public static styles = unsafeCSS(progressBarStyles);
+
+  /** Where the verification head is at */
+  @property({ attribute: "history-head-offset", type: Number })
+  public historyHeadOffset = 0;
```
[the remark this answers is not in the document]
Comment (hudson-newey) [reply src/components/progress-bar/progress-bar.ts:16]: Yeah, `historyHead` is an offset relative to the completed head

[the remark this answers is not in the document]
Comment (atruskie) [reply src/components/progress-bar/progress-bar.ts:16]: oh weird. I'd have probably just modelled it as two counting from 0 numbers

[the remark this answers is not in the document]
Comment (hudson-newey) [reply src/components/progress-bar/progress-bar.ts:16]: Yeah, I too thought it was a bit weird when I came across it.

My thoughts were: I shouldn't be refactoring how history works in this PR, and we're going to remove the double buffer in a following issue as well

[the remark this answers is not in the document]
Comment (atruskie) [reply src/components/progress-bar/progress-bar.ts:16]: true, but you can supply a number to this component however you like

**Review by nellihol — CHANGES_REQUESTED** [review]
Hi Hudson, looks great! 


Some questions/suggestions:  
1. Remove border
2. Change the main background color from transparent/white to ` --oe-panel-color`
3.  Make the progress bar thinner (in my mock-up it's 10px, use a variable that suits)
4. Make the progress head segment rounded e.g. `.progress-bar > .head-segment { border-radius: var(--oe-border-rounding);`
5. Make sure the progress bar is same width as the buttons above it.  So instead of 100% of the full container I think it would look more balanced as 100% of the button console, if that makes sense?
6. Make sure you have the same spacing between the progress bar and the "Do all of the audio segments..." instructive text to make it look balanced

Here is a mock-up for my suggestions: 
image

**Comment by hudson-newey** [comment]
I have updated the design with from requested feedback

I have added the revised designs to the PR's description

**Review by nellihol — APPROVED** [review]

**Pull request merged**